cashLetter.go

```diff
@@ -75,22 +76,38 @@ func (cl *CashLetter) setRecordType() {
 func (cl *CashLetter) Validate() error {
 	if cl.CashLetterHeader.RecordTypeIndicator == "N" {
 		if cl.GetBundles() != nil {
-			msg := fmt.Sprintf(msgCashLetterBundleEntries, cl.CashLetterHeader.RecordTypeIndicator)
-			return &CashLetterError{CashLetterID: cl.CashLetterHeader.CashLetterID,
-				FieldName: "RecordTypeIndicator", Msg: msg}
+			return &CashLetterError{
+				CashLetterID: cl.CashLetterHeader.CashLetterID,
+				FieldName:    "RecordTypeIndicator",
+				Msg:          fmt.Sprintf(msgCashLetterBundleEntries, cl.CashLetterHeader.RecordTypeIndicator),
+			}
 		}
 	}
 	switch cl.CashLetterHeader.CollectionTypeIndicator {
 	case
 		"00", "01", "02":
 	default:
 		if cl.GetRoutingNumberSummary() != nil {
-			msg := fmt.Sprintf(msgCashLetterRoutingNumber, cl.CashLetterHeader.CollectionTypeIndicator)
-			return &CashLetterError{CashLetterID: cl.CashLetterHeader.CashLetterID,
-				FieldName: "CollectionTypeIndicator", Msg: msg}
+			return &CashLetterError{
+				CashLetterID: cl.CashLetterHeader.CashLetterID,
+				FieldName:    "CollectionTypeIndicator",
+				Msg:          fmt.Sprintf(msgCashLetterRoutingNumber, cl.CashLetterHeader.CollectionTypeIndicator),
+			}
+		}
+	}
+
+	if cl.CashLetterControl == nil {
+		return &CashLetterError{
+			CashLetterID: cl.CashLetterHeader.CashLetterID,
+			FieldName:    "CashLetterControl",
+			Msg:          msgMandatoryRecord,
 		}
 	}
 
+	if err := cl.CashLetterControl.Validate(cl.CashLetterHeader.CollectionTypeIndicator); err != nil {
```

Comment:
CashLetterControl doesn't have a `CollectionTypeIndicator` field. I assume it's passed in this way because validation logic is dependent on this value, but it's not part of this record.

Reply:
Oh okay. What field is required now? The FI name?